### ISSUE TITLE
3D Plot tests fail

### ISSUE THREAD

**Issue by TheChymera** [IBT-FMI/SAMRI#82]
even after your latest PR (a505fc23b74ec7bd48da251921db361502bd09d6), the issue persists:

```
chymera@localhost ~/src/SAMRI/samri/plotting/tests $ pytest test_3D.py 
============================================= test session starts ==============================================
platform linux -- Python 3.6.6, pytest-4.2.0, py-1.5.4, pluggy-0.8.1
rootdir: /home/chymera/src/SAMRI, inifile:
collected 1 item                                                                                               

test_3D.py F                                                                                             [100%]

=================================================== FAILURES ===================================================
_________________________________________________ test_stat3D __________________________________________________

    def test_stat3D():
    	import samri.plotting.maps as maps
    
    	stat_map = "/usr/share/mouse-brain-atlases/dsurqec_200micron_roi-dr.nii"
    	template = "/usr/share/mouse-brain-atlases/dsurqec_40micron_masked.nii"
    
    	maps.stat3D(stat_map,
    		template=template,
    		save_as="_stat3D.png",
    		show_plot=False,
    		threshold=0.5,
>   		threshold_mesh = 0.5,
    		)

test_3D.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../maps.py:641: in stat3D
    plot_3D = _create_3Dplot(stat_maps,template_mesh=template_mesh,threshold=threshold_mesh,pos_values=pos_values,vmin=vmin,vmax=vmax)
../maps.py:501: in _create_3Dplot
    mesh = plt.imread("/var/tmp/3Dplot.png")
/usr/lib64/python3.6/site-packages/matplotlib/pyplot.py:2381: in imread
    return _imread(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

fname = '/var/tmp/3Dplot.png', format = None

    def imread(fname, format=None):
        """
        Read an image from a file into an array.
    
        *fname* may be a string path, a valid URL, or a Python
        file-like object.  If using a file object, it must be opened in binary
        mode.
    
        If *format* is provided, will try to read file of that type,
        otherwise the format is deduced from the filename.  If nothing can
        be deduced, PNG is tried.
    
        Return value is a :class:`numpy.array`.  For grayscale images, the
        return array is MxN.  For RGB images, the return value is MxNx3.
        For RGBA images the return value is MxNx4.
    
        matplotlib can only read PNGs natively, but if `PIL
        <http://www.pythonware.com/products/pil/>`_ is installed, it will
        use it to load the image and return an array (if possible) which
        can be used with :func:`~matplotlib.pyplot.imshow`. Note, URL strings
        may not be compatible with PIL. Check the PIL documentation for more
        information.
        """
    
        def pilread(fname):
            """try to load the image with PIL or return None"""
            try:
                from PIL import Image
            except ImportError:
                return None
            with Image.open(fname) as image:
                return pil_to_array(image)
    
        handlers = {'png': _png.read_png, }
        if format is None:
            if isinstance(fname, six.string_types):
                parsed = urlparse(fname)
                # If the string is a URL, assume png
                if len(parsed.scheme) > 1:
                    ext = 'png'
                else:
                    basename, ext = os.path.splitext(fname)
                    ext = ext.lower()[1:]
            elif hasattr(fname, 'name'):
                basename, ext = os.path.splitext(fname.name)
                ext = ext.lower()[1:]
            else:
                ext = 'png'
        else:
            ext = format
    
        if ext not in handlers:
            im = pilread(fname)
            if im is None:
                raise ValueError('Only know how to handle extensions: %s; '
                                 'with Pillow installed matplotlib can handle '
                                 'more images' % list(handlers))
            return im
    
        handler = handlers[ext]
    
        # To handle Unicode filenames, we pass a file object to the PNG
        # reader extension, since Python handles them quite well, but it's
        # tricky in C.
        if isinstance(fname, six.string_types):
            parsed = urlparse(fname)
            # If fname is a URL, download the data
            if len(parsed.scheme) > 1:
                fd = BytesIO(urlopen(fname).read())
                return handler(fd)
            else:
>               with open(fname, 'rb') as fd:
E               FileNotFoundError: [Errno 2] No such file or directory: '/var/tmp/3Dplot.png'

/usr/lib64/python3.6/site-packages/matplotlib/image.py:1375: FileNotFoundError
--------------------------------------------- Captured stdout call ---------------------------------------------
0.5
File written 30%
File written 60%
Color management: using fallback mode for management
Read prefs: /home/chymera/.config/blender/2.79/config/userpref.blend
Error: Cannot write a single file with an animation format selected
#ff7e00
(  0.0000 sec |   0.0000 sec) Importing OBJ '/usr/share/mouse-brain-atlases/ambmc2dsurqec_15micron_masked.obj'...
  (  0.0001 sec |   0.0000 sec) Parsing OBJ file...
    (  6.3095 sec |   6.3094 sec) Done, loading materials and images...
    (  6.3096 sec |   6.3095 sec) Done, building geometries (verts:312399 faces:624646 materials: 0 smoothgroups:1) ...
    ( 13.0193 sec |  13.0192 sec) Done.
  ( 13.0194 sec |  13.0193 sec) Finished importing: '/usr/share/mouse-brain-atlases/ambmc2dsurqec_15micron_masked.obj'
Progress: 100.00%

(  0.0000 sec |   0.0000 sec) Importing OBJ '/var/tmp/dsurqec_200micron_roi-dr_pos_mesh.obj'...
  (  0.0001 sec |   0.0001 sec) Parsing OBJ file...
    (  0.0022 sec |   0.0021 sec) Done, loading materials and images...
    (  0.0022 sec |   0.0022 sec) Done, building geometries (verts:140 faces:276 materials: 0 smoothgroups:0) ...
    (  0.0046 sec |   0.0045 sec) Done.
  (  0.0046 sec |   0.0046 sec) Finished importing: '/var/tmp/dsurqec_200micron_roi-dr_pos_mesh.obj'
Progress: 100.00%

Error: Cannot write a single file with an animation format selected

Blender quit
--------------------------------------------- Captured stderr call ---------------------------------------------
Traceback (most recent call last):
  File "/usr/share/blender/2.79/scripts/modules/addon_utils.py", line 331, in enable
    mod = __import__(module_name)
ModuleNotFoundError: No module named 'cycles'
Traceback (most recent call last):
  File "/home/chymera/src/SAMRI/samri/plotting/blender_visualization.py", line 264, in <module>
    TakePic("3Dplot")
  File "/home/chymera/src/SAMRI/samri/plotting/blender_visualization.py", line 239, in TakePic
    bpy.ops.render.render( write_still=True )
  File "/usr/share/blender/2.79/scripts/modules/bpy/ops.py", line 189, in __call__
    ret = op_call(self.idname_py(), None, kw)
RuntimeError: Error: Cannot write a single file with an animation format selected

---------------------------------------------- Captured log call -----------------------------------------------
due.py                      63 ERROR    Failed to import duecredit due to No module named 'duecredit'
=============================================== warnings summary ===============================================
samri/plotting/tests/test_3D.py::test_stat3D
samri/plotting/tests/test_3D.py::test_stat3D
samri/plotting/tests/test_3D.py::test_stat3D
  /usr/lib64/python3.6/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
    return f(*args, **kwds)

samri/plotting/tests/test_3D.py::test_stat3D
  /usr/lib64/python3.6/site-packages/scipy/ndimage/measurements.py:272: DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    return _nd_image.find_objects(input, max_label)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================== 1 failed, 4 warnings in 22.77 seconds =====================================
```

Are you sure this is not an issue with blender USE flags?

```
[I] media-gfx/blender
     Available versions:  2.72b-r4 (~)2.79 2.79-r1 (~)2.79b-r1 {+boost +bullet collada colorio cuda cycles +dds debug doc +elbeem ffmpeg fftw +game-engine headless jack jemalloc jpeg2k libav llvm man ndof nls openal opencl +openexr openimageio openmp +opennl opensubdiv openvdb osl player redcode sdl sndfile test tiff valgrind CPU_FLAGS_X86="sse sse2" PYTHON_SINGLE_TARGET="python3_5 python3_6" PYTHON_TARGETS="python3_4 python3_5 python3_6"}
     Installed versions:  2.79b-r1(11:21:47 PM 12/19/2018)(bullet dds elbeem ffmpeg game-engine nls openal openexr openmp sdl tiff -collada -colorio -cuda -cycles -debug -doc -fftw -headless -jack -jemalloc -jpeg2k -libav -llvm -man -ndof -opencl -openimageio -opensubdiv -openvdb -osl -player -sndfile -test -valgrind PYTHON_SINGLE_TARGET="python3_6 -python3_5" PYTHON_TARGETS="python3_6 -python3_5")
     Homepage:            http://www.blender.org
     Description:         3D Creation/Animation/Publishing System
```

**Comment by setina42** [comment]
Maybe. What do you have in `package.use` regarding blender? I have the following USE flags:
`media-gfx/blender cycles tiff openimageio
`

**Comment by TheChymera** [comment]
@setina42 `/etc/portage/package.use` just provides a subset of the requirements. To see all flags, please use `eix`.

The bug happens both with the flags I pasted above and with:

```
[I] media-gfx/blender
     Available versions:  2.72b-r4 (~)2.79 2.79-r1 (~)2.79b-r1 {+boost +bullet collada colorio cuda cycles +dds debug doc +elbeem ffmpeg fftw +game-engine headless jack jemalloc jpeg2k libav llvm man ndof nls openal opencl +openexr openimageio openmp +opennl opensubdiv openvdb osl player redcode sdl sndfile test tiff valgrind CPU_FLAGS_X86="sse sse2" PYTHON_SINGLE_TARGET="python3_5 python3_6" PYTHON_TARGETS="python3_4 python3_5 python3_6"}
     Installed versions:  2.79b-r1(09:58:54 PM 02/17/2019)(bullet cycles dds elbeem ffmpeg game-engine nls openal openexr openimageio openmp sdl tiff -collada -colorio -cuda -debug -doc -fftw -headless -jack -jemalloc -jpeg2k -libav -llvm -man -ndof -opencl -opensubdiv -openvdb -osl -player -sndfile -test -valgrind PYTHON_SINGLE_TARGET="python3_6 -python3_5" PYTHON_TARGETS="python3_6 -python3_5")
     Homepage:            http://www.blender.org
     Description:         3D Creation/Animation/Publishing System
```

**Comment by TheChymera** [comment]
Ok, this seems was cause by my blender configuration. As I use it mainly for video editing, it seems the mesh-creation function inherits the environment.

```
rm -rf ~/.config/blender/
```

as my user solved it, but obviously we can't rely on this being done. Is there any way for you to unload the user config in the python interface?
